### DISCUSSION TITLE
Third lightbulb for 02-TwoSimpleLightBulbs

### DISCUSSION THREAD

**Discussion by pauliu749** [HomeSpan/HomeSpan#405] [Q&A] [answered]
I tried 02-TwoSimpleLightBulbs can be pair with Homekit without problem, two lightbulbs show on Homekit correctly, but when I add third lightbulb by copying the program lines from first lightbulb, homekit will show error after added an unexpected Bridge and lightbulb2. 

  new SpanAccessory();             
  
    new Service::AccessoryInformation(); 
      new Characteristic::Identify();       

    new Service::LightBulb();         
      new Characteristic::On();    

Actually Bridge did not appear on original 02-TwoSimpleLightBulbs.

How to add third lightblub correctly?

Thanks for any advise :)

**Comment by HomeSpan** [comment, marked as answer]
Hmmm.  It seems that the latest version of the Home App has trouble pairing with a device that has more than two lightbulbs unless the device is configured as a bridge.  I don't know if this is something I never noticed before, or if it is a new limitation in HomeKit, but I just tested this and have the same issue.

Fortunately the solution is to simply configure HomeSpan as a Bridge device.  See [Example 08-Bridges](https://github.com/HomeSpan/HomeSpan/tree/master/examples/08-Bridges) for how to do this (you basically create an initial SpanAccessory with nothing but the AccessoryInformation Service).  When configured as a Bridge you should be able to add dozens of lightbulbs without any issues.

Note: When making significant changes to a device that has already been paired, I find it best to start fresh.  First, unpair the device via the Home App by selecting "Remove" from the setting screen under the Home App tile for the Accessory.  The "Remove" option is often found under the "Bridge" submenu, which is poorly named for devices that are not actually bridges (in prior versions of the Home App, the Bridge sub-menu only appeared if the device was configured as a Bridge).  Then, type "H" into the Serial Monitor of the Arduino IDE --- this reboots the device and generates a fresh HomeKit Device ID so that the Home App won't confuse it with the previous configuration.  When you re-pair the device from the Home App, it should recognize the device as a Bridge, and then step you through adding each light bulb.

**Reply by pauliu749** [reply]
Thanks for the advise, it solved my problem 😊
